### PR TITLE
tiny JS world --done

### PR DESCRIPTION
**Pull request by korotushak96**
_Replace task name and links as appropriate and remove this line._

### a-tiny-JS-world

[Demo](https://korotushak96.github.io/a-tiny-JS-world/) |
[Code base](https://github.com/korotushak96/a-tiny-JS-world)

The code is submitted in its own sub-directory and in a dedicated feature branch.
Please, review.

### CODE REVIEW

File: submissions/korotushak96/a-tiny-JS-world/index.js [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+/* Print examples:
+   print('ABC');
+   print('<strong>ABC</strong>');
+   print('<strong>ABC</strong>', 'div');
+
+   print('human; John; male; 2; 2; Hello world!; Rex, Tom, Jenny');
+   print('human; <strong>John</strong>; male; 2; 2; <em>Hello world!</em>; Rex, Tom, Jenny');
+   print('human; <strong>John</strong>; male; 2; 2; <em>Hello world!</em>; Rex, Tom, Jenny', 'div');
+   */
+  const dog = {
```
Comment (o-msh) [inline submissions/korotushak96/a-tiny-JS-world/index.js]: Please remove redundant spaces. You code should be pretty. Also you can use IDE features for nice code )

File: submissions/korotushak96/a-tiny-JS-world/index.js [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+    name: 'Jess',
+    gender: 'female',
+    legs: 2,
+    hands: 2,
+    clawed: true,
+    saying: cat.saying
+};
+
+[dog, cat, woman, man, catWoman].forEach(el => {
+    print(`Hi! I am ${el.species}; My name is <strong>${el.name}</strong>;
```
Comment (o-msh) [inline submissions/korotushak96/a-tiny-JS-world/index.js]: There is a pattern here. Please, implement a function that would take an object as an argument and return a formatted string where components are combined employing `Array.join`.
After that you code look like `[dog, cat, woman, man, catWoman].forEach(el => print(getParameters(el)))`

File: submissions/korotushak96/a-tiny-JS-world/index.js (hunk 1 above)
Comment (korotushak96) [reply submissions/korotushak96/a-tiny-JS-world/index.js]: I'm going to make my self to do it ;)

File: submissions/korotushak96/a-tiny-JS-world/index.js [hunk 3]
```diff
@@ -73,8 +73,12 @@ const catWoman = {
     saying: cat.saying
 };
 
+let getParameters = ({species, name, gender, legs, hands, saying}) =>{
```
Comment (o-msh) [inline submissions/korotushak96/a-tiny-JS-world/index.js]: Please, use `const` for define functions.

File: submissions/korotushak96/a-tiny-JS-world/index.js [hunk 4]
```diff
@@ -27,32 +27,32 @@
    print('human; <strong>John</strong>; male; 2; 2; <em>Hello world!</em>; Rex, Tom, Jenny');
    print('human; <strong>John</strong>; male; 2; 2; <em>Hello world!</em>; Rex, Tom, Jenny', 'div');
    */
-  const dog = {
+const dog = {
     species: 'dog',
```
Comment (o-msh) [inline submissions/korotushak96/a-tiny-JS-world/index.js]: species: `dog`, `cat` and `human` are repeated. You can define object for constants.

Comment (o-msh) [reply submissions/korotushak96/a-tiny-JS-world/index.js]: Also you can define as constants some other repeated parameters, how about `gender` ?

File: submissions/korotushak96/a-tiny-JS-world/index.js [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+   print('<strong>ABC</strong>', 'div');
+
+   print('human; John; male; 2; 2; Hello world!; Rex, Tom, Jenny');
+   print('human; <strong>John</strong>; male; 2; 2; <em>Hello world!</em>; Rex, Tom, Jenny');
+   print('human; <strong>John</strong>; male; 2; 2; <em>Hello world!</em>; Rex, Tom, Jenny', 'div');
+   */
+const dog = {
+    species: 'dog',
+    name: 'Dina',
+    gender: 'female',
```
Comment (o-msh) [inline submissions/korotushak96/a-tiny-JS-world/index.js]: Try define constants for repeated properties

File: submissions/korotushak96/a-tiny-JS-world/index.js (hunk 3 above)
Comment (korotushak96) [reply submissions/korotushak96/a-tiny-JS-world/index.js]: I add a few, but I don't undrestand reasons in this case. Please, explaine.

File: submissions/korotushak96/a-tiny-JS-world/index.js (hunk 5 above)
Comment (korotushak96) [reply submissions/korotushak96/a-tiny-JS-world/index.js]: I add classes. But I'm not sure I used correct inheritance. Please, check it

File: submissions/korotushak96/a-tiny-JS-world/index.js (hunk 3 above)
Comment (korotushak96) [reply submissions/korotushak96/a-tiny-JS-world/index.js]: okay ;)

File: submissions/korotushak96/a-tiny-JS-world/index.js (hunk 5 above)
Comment (o-msh) [reply submissions/korotushak96/a-tiny-JS-world/index.js]: @korotushak96 You don't need use ES6 Class here. Please, return to simple `Object` definition. You will improve this task with ES6 Class in future.

Comment (o-msh) [reply submissions/korotushak96/a-tiny-JS-world/index.js]: IMO looks like this.
```javascript
const INHABITANT = {
    species: {
        dog: 'dog',
        human: 'human',
        ...etc
    },
    genders: {
        male: 'male',
        female: 'female',
    }
};
const dog = {
    name: "Paolo",
    species: INHABITANT.species.dog,
    gender: INHABITANT.genders.male,
    legs: 4,
    hands: 0,
    saying: "Bark",
};
...etc
```

**Comment by o-msh** [comment]
@korotushak96 Do you need any help with you task?

File: submissions/korotushak96/a-tiny-JS-world/index.js (hunk 5 above)
Comment (korotushak96) [reply submissions/korotushak96/a-tiny-JS-world/index.js]: ok =) done

**Comment by korotushak96** [comment]
No. I don't =) Sorry for my absence

**Pull request merged**